src/thought/view-thoughts/thoughts-mind-map-view.js

```diff
@@ -1,281 +1,286 @@
 
-  var parent = thoughtStorage.findThoughtById(parentId);
+    this.setState({currentViewedThoughtId: targetThoughtId});
+    thoughtStorage.logTree();
+    var targetThought = thoughtStorage.findThoughtById(targetThoughtId);
 
-  return thoughtStorage.create(thought, parent)
-    .then(function (newThought) {
-      thought.id = newThought.id;
+    if (!targetThought) {
+      throw new Error('changeThought(): couldn\'t find targetThought in thoughtStorage by id: ', targetThoughtId);
+    }
+    console.debug('thoughts-mind-map-view.thoughtClickHandler(): targetThought: ', targetThought);
 
-      var children = [thought];
+    this.changeThought(targetThought);
+  };
 
-      thoughtStorage.addChildrenToTree({
-        parentId: parentId,
-        children: children
-      });
+  visNetworkClickHandler = event => {
+    this.closeThoughtNameEditor();
+    if (visNetworkHelper.clickedOnThought(event)) {
+      console.debug('change thought!');
+      console.debug('event: ', event);
+      let targetThoughtId = visNetworkHelper.getTargetThoughtId(event);
 
-      brainVisNetwork.addChildThoughts({
-        children: children,
-        parentThoughtId: parentId
-      });
+      this.thoughtClickHandler(targetThoughtId);
+    }
+  };
 
-      editThought(thought.id);
+  visNetworkDoubleClickHandler = event => {
+    if (visNetworkHelper.clickedOnThought(event)) {
+      let targetThoughtId = visNetworkHelper.getTargetThoughtId(event);
+      console.info('=== Event === Node double click');
+      this.createEmptyChild(targetThoughtId);
+    }
+  };
 
-      return thought;
-    });
-}
+  visNetworkHoldHandler = event => {
+    if (visNetworkHelper.clickedOnThought(event)) {
+      let targetThoughtId = visNetworkHelper.getTargetThoughtId(event);
+      console.info('=== Event === Node hold');
+      this.editThought(targetThoughtId);
+    }
+  };
 
-function thoughtClickHandler(targetThoughtId) {
-  // if clicking on the current thought, do nothing.
-  if (targetThoughtId === currentViewedThoughtId) {
-    console.info('Click was on the selected note, doing nothing')
-    return;
-  }
+  editThought(targetThoughtId) {
+    const thought = thoughtStorage.findThoughtById(targetThoughtId);
 
-  var currentViewedThoughtId_temp = _.find(brainVisNetwork.visNodes.getIds(),
-    function (nodeId) {
-      return nodeId == targetThoughtId;
+    if (thought.name === thoughtStorage.APP_FOLDER_NAME) {
+      console.info('It is not allowed to edit App root folder name');
+      return;
     }
-  );
 
-  console.debug('currentViewedThoughtId from visNodes: ', currentViewedThoughtId_temp);
+    if (!thought.isNote) {
+      console.info('It is not allowed to edit user-uploaded files');
+      return;
+    }
 
-  var node = brainVisNetwork.visNodes.get(currentViewedThoughtId_temp);
-  console.debug('node from visNodes: ', node);
+    this.setState({
+      currentViewedThought: thought,
+      editNote: {
```

Comment:
`editNote` is basically the same thing as the name of the method: `editThought` (after rename it will be `editNote()`).

So this should be `editedNote` or `noteToEdit` or `editorNote` or `noteForEditor`. And actually it's the same as selected note, so maybe makes sense to just use `selectedNote` here in order to simplify the naming problem.

Reply:
renamed to selectedNote